dashboard/app/models/course.rb

```diff
@@ -121,4 +121,57 @@ def summarize
       end
     }
   end
+
+  @@course_cache = nil
+  COURSE_CACHE_KEY = 'course-cache'.freeze
+
+  def self.should_cache?
+    return false if Rails.application.config.levelbuilder_mode
+    ENV['UNIT_TEST'] || ENV['CI']
+  end
+
+  # generates our course_cache from what is in the Rails cache
+  def self.course_cache_from_cache
+    Course.connection
```

Comment:
Looks like it maybe ensures that a connection is established for this thread, but that should happen anyway in the below query. It doesn't appear to do anything special for the method. See http://api.rubyonrails.org/classes/ActiveRecord/Base.html